src/main/java/it/smartcommunitylab/aac/crypto/CertificateParser.java

```diff
@@ -0,0 +1,104 @@
+package it.smartcommunitylab.aac.crypto;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.springframework.security.saml2.core.Saml2X509Credential;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.PrivateKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class CertificateParser {
+
+    // updatePemHeaderFooter ensured that a PEM (as UTF-8 string) starts
+    // and terminated with the appropriate footer. Which footer is appropriate
+    // depends in the the params kind
+    // TODO: kind should be enum
+    private static String updatePemHeaderFooter(String value, String kind) {
```

Comment:
drop the methods manipulating user input. We can try guessing content if needed for parsing, but we won't fix/correct user input